### ISSUE TITLE
Does not support "." (dots) in profile names

### ISSUE THREAD

**Issue by PhilLavoie** [smithy-lang/smithy-typescript#1163]
Given this ~/.aws/config INI:
```ini
[profile my.big.profile]
region = us-east-1
sso_session = my.sso
sso_account_id = 123456789
sso_role_name = BigRole

[sso-session my.sso]
sso_region = us-east-1
sso_start_url = https://magic-app.awsapps.com/start
sso_registration_scopes = sso:account:access
```
The [loadSsoSessionData](https://github.com/smithy-lang/smithy-typescript/blob/main/packages/shared-ini-file-loader/src/getSsoSessionData.ts#L14) returns an object where the key is `my` instead of `my.sso`. This breaks the [fromSso](https://github.com/aws/aws-sdk-js-v3/blob/main/packages/credential-provider-sso/src/fromSSO.ts#L104) function which tries to get the session data using `my.sso`, literally.

This is a limitation of the AWS SDK for JS v3, as the `aws` CLI and the golang SDK support SSO sessions with dots in their names just fine.

**Comment by PhilLavoie** [comment]
It seems like the fix would be to change that [line](https://github.com/smithy-lang/smithy-typescript/blob/main/packages/shared-ini-file-loader/src/getSsoSessionData.ts#L14) for:
```ts
.reduce((acc, [key, value]) => ({ ...acc, [key.split(CONFIG_PREFIX_SEPARATOR).slice(1).join(CONFIG_PREFIX_SEPARATOR)]: value }), {});
```

**Comment by kuhe** [comment]
@trivikr says the fix was released in [@smithy/shared-ini-file-loader@2.2.3](https://github.com/smithy-lang/smithy-typescript/blob/3357fda448ea8cee3f7bfa7caeda2768a9751f8f/packages/shared-ini-file-loader/CHANGELOG.md#223)

**Comment by trivikr** [comment]
> fix was released in [@smithy/shared-ini-file-loader@2.2.3](https://github.com/smithy-lang/smithy-typescript/blob/3357fda448ea8cee3f7bfa7caeda2768a9751f8f/packages/shared-ini-file-loader/CHANGELOG.md#223)

The feature request was implemented in `parseIni` in https://github.com/smithy-lang/smithy-typescript/pull/1067
However, it was not moved to other utilities which consume parseIni, like loadSsoSessionData.

We'll have to get a minimal repro, and investigate what role `CONFIG_PREFIX_SEPARATOR` plays.
